### ISSUE TITLE
Minibatch sequence length

### ISSUE THREAD

**Issue by r-kellerm** [nadavbra/protein_bert#51]
Hi,

First, thank you very much for making the code and the pretrained weight available and so user-friendly.

Wanted to ask about the minibatch data preparation -
In the paper (section 2.2) you write: "When encoding a protein longer than the chosen sequence length, we selected a random subsequence of the protein, leaving out at least one of its two ends."
However [in the code](https://github.com/nadavbra/protein_bert/blob/443129f062043ba44e3fdcb97ca7b47627861db0/proteinbert/finetuning.py#L219) you filter samples that exceed the current sequence length, but you still add the START and END tokens.
Is this dismissed on purpose?

Thanks!

**Comment by nadavbra** [comment]
@r-kellerm

The sentence you quote refers to the pretraining of ProteinBERT. For fine-tuning, we did filter sequences by length:

> Throughout the entire fine-tuning process we used a sequence length of 512 tokens, except for a final epoch of 1,024 tokens which was introduced to encourage the model to generalize to different sequence lengths. For all but the last epoch, we filtered out all the proteins larger than 512 tokens from the training and validation set, and trained the model on the remaining records. For the final epoch, we trained the model on records from the training and validation set of size up to 1,024 tokens.

(see "Fine-tuning and evaluation" in the Supplementary Methods for more details)

Since ProteinBERT is pretrained on all sequence lengths and includes different sequence lengths in fine-tuning, we show in our paper that it can generalize to sequence lengths that aren't encountered during fine-tuning (see Figure 4 in the paper).

In all cases, whether or not sequences are filtered by length, _START_ and _END_ tokens must be included **prior** to sub-sequencing.

**Comment by r-kellerm** [comment]
Thank you very much for the clarification!
